aten/src/ATen/native/vulkan/api/Common.h

```diff
@@ -48,6 +49,12 @@ struct Resource;
 class Runtime;
 struct Shader;
 
+struct GPU final {
+  const Adapter* adapter;
```

Comment:
Is it ok to have here pointer and not reference, will it be valid to have nullptr adapter?

Reply:
Great point, thanks.  Having a reference will make this class non-copyable.  I think in this particular case it might be a problem considering my use of this class elsewhere, but I will double check to see if that is the case.  You have a solid point there though that references are superior to pointers.